### PR TITLE
Add ability to maximize and minimize workspace

### PR DESCRIPTION
**Pull request by donaldkibet**
### What does this PR do?

1. Add ability to minimize, maximize and hide content workspace as shown in this [design](https://app.zeplin.io/project/5f7223cfda10f94d67cec6d0/screen/60b8b3bfae7d2da4cee53a32) 

### Screenshoot

workspace

### CODE REVIEW

File: packages/esm-patient-chart-app/src/workspace/context-workspace.component.tsx [hunk 1]
```diff
@@ -10,30 +12,83 @@ import { useWorkspace } from '../hooks/useWorkspace';
 import { patientChartWorkspaceSlot } from '../constants';
 import { isDesktop } from '../utils';
 
+type WindowSize = 'maximize' | 'normal' | 'hide';
+type ActionTypes = 'minimize' | 'maximize' | 'hide' | 'reopen';
+
+const reducer = (state: WindowSize, action: ActionTypes) => {
+  switch (action) {
+    case 'maximize':
+      return (state = 'maximize');
```
Comment (FlorianRappl) [inline packages/esm-patient-chart-app/src/workspace/context-workspace.component.tsx:21]: Hm this assignment is rather confusing. Why not just `return 'maximize'`? Same for the rest.

**Review by FlorianRappl — COMMENTED** [review]
LGTM - great job!

**Review by FlorianRappl — APPROVED** [review]

**Comment by jonathandick** [comment]
Is it complicated to make this resizable by dragging the edge to the desired width?

**Comment by donaldkibet** [comment]
> Is it complicated to make this resizable by dragging the edge to the desired width?

looking into that

**Pull request merged**